protocol/messenger_contacts.go

```diff
@@ -333,19 +306,14 @@ func (m *Messenger) updateAcceptedContactRequest(response *MessengerResponse, co
 	response.AddMessage(contactRequest)
 	response.AddContact(contact)
 
-	// Remove last sent system message
-	err = m.removeLastMutualStateUpdateMessage(response, contact.ID)
-	if err != nil {
-		return nil, err
+	// Add mutual state update message for incoming contact request
+	chat, ok := m.allChats.Load(contact.ID)
+	if !ok {
```

Comment:
@MishkaRogachev I think it would be better to move this check right after the `contact` variable is defined in the beginning of the function. This way, we avoid unnecessary side-effects that can happen between the definition of the variable on line 255 and the chat existence check on line 311.

The idea being: load as much data as possible in the beginning of the function and validate that, and only then proceed with the side-effects. This is a practice I like for status-go in particular, because in this repository, when things fail in the middle of processes, we usually don't attempt to rollback anything, we just short-circuit and return an error, leaving the system in a potentially inconsistent state. Eventual consistency seems to be our preferred approach, but some inconsistencies could be avoided altogether.

For example, if we validate the chat exists right after loading the `contact` in memory and if the check fails, then `dispatchMessage` would never be called and no notification would be needlessly generated by `addActivityCenterNotification`.

The same idea could be used to move the line that does `GetActivityCenterNotificationByID` to the beginning of the function, then the check for the existence of `notification` would happen earlier and this would prevent some side-effects.

My comment here is leaning more on the side of a general good practice for status-go, not so much for your PR. You may disagree of course :) Feel free to ignore my suggestion.

Reply:
@ilmotta thanks for the tip, already fixed it!